py/lexer.c

```diff
@@ -291,15 +334,69 @@ STATIC void parse_string_literal(mp_lexer_t *lex, bool is_raw) {
     }
 
     size_t n_closing = 0;
+    # if MICROPY_PY_FSTRING
```

Comment:
There shouldn't be a space between the `#` and the `if` (same for endif below, and others below)